### PR TITLE
CASM-4875: Adds Rack Resiliency Command Line Interface Feature

### PR DESCRIPTION
**Pull request by arka-pramanik-hpe**
### Rack Resiliency Cray CLI Interface

[Design Link](https://rndwiki-pro.its.hpecorp.net/display/CASMPET/Rack+Resiliency+HLD)
[EPIC Link](https://jira-pro.it.hpe.com:8443/browse/CASM-4875)
[JIRA Link](https://jira-pro.it.hpe.com:8443/browse/CASM-5605)
[Dependency Link](https://github.com/Cray-HPE/cray-rrs)

Implementation of the Rack Resiliency Service (RRS) CLI interface, providing commands for managing zones, critical services, monitoring service status, and updating configurations. This feature enables administrators to manage rack resiliency through the Cray CLI, making it easier to monitor and configure critical services across zones.

The following commands have been implemented:

- `cray rrs zones list`: Lists all configured zones
- `cray rrs zones describe <zone-name>`: Gets detailed information about a specific zone
- `cray rrs criticalservices list`: Lists all critical services grouped by namespace
- `cray rrs criticalservices describe <critical-service-name>`: Gets summarized information about a specific critical service
- `cray rrs criticalservices status list`: Lists status of all critical services with distribution details
- `cray rrs criticalservices status describe <critical-service-name>`: Gets detailed status information about a specific critical service
- `cray rrs criticalservices update --from-file <file-path>`: Updates critical services configuration using a JSON file

This change is backwards compatible as it adds new functionality without modifying existing behaviors.

## Issues and Related PRs
* Resolves [CASM-4875](https://jira-pro.it.hpe.com:8443/browse/CASM-4875)

## Testing
Tested on:
* Internal development system
* Local development environment
* Surtur

Test description:
- Verified all CLI commands return expected results:
  - Zone listing and description
  - Critical services listing and description
  - Service status monitoring
  - Configuration updates
- Validated command output formats match specification
- Confirmed proper error handling for edge cases
- Install/upgrade validation checks were run using goss tests
- Continuous integration tests were executed successfully
- Upgrade was tested to ensure backward compatibility
- Downgrade testing was not applicable as this is a new feature addition

## Risks and Mitigations
No risks.

## Pull Request Checklist
- [ ] Version number(s) incremented
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [ ] CHANGELOG.md updated
- [x] Testing is appropriate and complete

### CODE REVIEW

**Comment by ravikanth-nalla-hpe** [comment]
Please squash all the commits.

File: cray/modules/rrs/swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,819 @@
...
+          - "No K8s Topology/Ceph Zones configured"
+          - "No Ceph zones configured"
+          - "No K8s topology zones configured"
+
+      ### GET /zones/{zone_name}
+        Retrieve detailed information for a specific zone including:
+          - Zone Name
+          - Management Masters
+          - Management Workers
+          - Management Storages
```
Comment (dborman-hpe) [inline cray/modules/rrs/swagger.yaml:55]: Up above in lines 41-43 it has `Management Master Nodes` etc., but here it is `Management Masters`.  Is that intentional to use two different names?

Also, we really should be switching away from calling them `Master` nodes, and instead calling them `Control Plane` nodes, that transition started back in k8s 1.20 and is was completed by k8s 1.27, we're now at k8s 1.32 in CSM 1.7.  One could use `Management Control Plane (Master) Nodes`, but maybe that's a bit wordy.  Anyway, use your judgement on whether or not you want to pick up the newer `Control Plane` term.

File: cray/modules/rrs/swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,819 @@
...
+        Retrieve a summarized view of a specific critical service (without pod details). The response includes:
+          - Configured Instances
+          - Currently Running Instances
+          - Name
+          - Namespace
+          - Type
+
+      ### PATCH /criticalservices
+        Update the critical services configuration based on a provided file input. This endpoint allows
+        you to modify how critical services are configured in the system
```
Comment (dborman-hpe) [inline cray/modules/rrs/swagger.yaml:71]: Does this replace the existing configuration, or just update it (which is what `patch` implies).

At a high level what I'm not seeing in this interface is:

1. How is the initial configuration set up?  By using `PATCH`, or some other mechanism?
2. Can I save the configuration to a file?
3. And then can I restore the configuration from that file?

#2 and #3 are basically backup/restore

File: cray/modules/rrs/swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,819 @@
...
+    information and detailed critical service status. It gathers node details across various zones
+    and presents both high-level summaries and in-depth information for zones and critical services.
+
+    ## Resources
+
+      ### GET /zones
+        Retrieve aggregated zone configuration showing Kubernetes Topology Zones and CEPH Zones including:
+          - Management Master Nodes
+          - Management Storage Nodes
+          - Management Worker Nodes
```
Comment (dborman-hpe) [inline cray/modules/rrs/swagger.yaml:43]: It seems other places have `Storage Nodes` listed third, not second.

**Review by dborman-hpe — APPROVED** [review]
LGTM

**Review by mharding-hpe — APPROVED** [review]
